baybe/simulation/lookup.py

```diff
@@ -159,3 +139,41 @@ def _look_up_targets_from_dataframe(
 
     # Add the lookup values
     queries.loc[:, target_names] = np.asarray(all_match_vals)
+
+
+def label_columns(
```

Comment:
There is the general question how we want to construct our lookup callables in our examples. In principle, we can also drop this utility and require people and us create the dataframe-based directly, which would save one level of complexity / code for this function. Disadvantage would be that:
* People will forget setting the proper dataframe indexes all the time! (happened to me every single time)
* The decorator gives you access to both the numpy and pandas version, which might be helpful for plotting, for example.

But I'm curious to hear how you would set up our examples

Reply:
I would say
1. All examples should be set up according to the new design, except for
2. Examples where we utilize botorch functions could be handled by a helper/wrapper. This has the advantage to also illustrate how to use simulation with lookups that have the old design

what do you mean by `People will forget setting the proper dataframe indexes all the time! (happened to me every single time)`

Reply:
See my comment in the example: I think if we now require a `Callable[[DataFrame], DataFrame]`, then this should actually be used in the example. Having a helper is nice, but this then also needs to be explained properly and not be the only thing shown in the example.

Reply:
@AdrianSosic still not sure what the index problem you describe is supposed to be, please clarify

Reply:
As said previously, the problem is that if you have an incoming df with a non-trivial index, and in the function you create a new dataframe from it, then you must remember setting its index since otherwise subsequent concatenation operations etc will fail.

Reply:
The problem appears, for instance, if you use a callable lookup for a state space with discrete component an run any of our simulation utilities.

Reply:
Is that something we care about or do we say "responsibility of the user to provide the correct format"? I can only share my experience here: forgot it every single time I played with code in the PR. However, unless we offer some automatic way to create that callable (like the decorator I had previously), there is basically no way to avoid it.

Reply:
where exactly does baybe/simulation throw the error if that happens?
The better solution is to fix this and make indices irrelevant, but its still not clear to me what causes the error

Reply:
I think it is fixed by changing `_look_up_targets_from_callable`
from 

```python
    df_targets = lookup(queries)
    queries[df_targets.columns] = df_targets
```

to
```python
    df_targets = lookup(queries)
    queries[df_targets.columns] = df_targets.values
```

commit suggestion made here https://github.com/emdgroup/baybe/pull/441#discussion_r1891430190

Reply:
Seems like 👍🏼  Even though I don't think it is technically correct because now we effectively ignore the index. So this brings us into the situation where in some places we say "the index definitely matters!" and in others we say "index doesn't matter". But ok, let's be pragmatic for now, since the simulation module is largely still a mess anyway + I hope that we can switch to polars at some point entirely

Reply:
where do we say the index matters

Reply:
For example our recommendations in discrete/hybrid spaces carry the index of the discrete space

Reply:
this is true, but in no way does imply `the index definitely matters`. It is provided as mere convenience and I think we could simply drop the line or reset the index when returning and nothing would/should change

I think our code should do the exact opposite: should not care about index anywhere, hence I also don't agree that this fix is `technically incorrect`. With the change the only thin that the user needs to do is provide the targets int he same row-order as the input - which imo is a requirement that is rather obvious

Reply:
Don't get me wrong, I'm entirely with you: the code should not rely on the index! And that will ultimately even be the requirement to make a potential polars transition possible. The recommendation batch was just the example that came immediately to my mind. My point is simply: if you ask me if our entire codebase is index-agnostic, I would not be 100% confident to answer with "yes". But I'll resolve this now, OK. I think we are aligned 👍🏼

Reply:
of course the entire code base is not index agnostic - after all we are using pandas
but what matters is that the user-facing input of measurements and experiments is fully index agnostic, if that is not the case we need to fix it so if you know more problematic places please point them out

Reply:
yes, that's what I meant. I'm just not sure at the moment if there at no other such places. I suspect especially some of them in the simulation module. But we'll fix them along the way ... 🐧